src/main/java/com/spotify/confidence/EventSenderEngineImpl.java

```diff
@@ -58,26 +59,26 @@ public void run() {
         }
       }
     }
-  }
 
-  private void scheduleFlush() {
-    // Cancel the existing scheduled task if it exists
-    if (pendingFlush != null && !pendingFlush.isDone()) {
-      pendingFlush.cancel(false);
-    }
-    if (flushTimeoutMilliseconds > 0) {
-      pendingFlush =
-          executorService.schedule(this::flush, flushTimeoutMilliseconds, TimeUnit.MILLISECONDS);
+    private void scheduleFlush() {
+      // Cancel the existing scheduled task if it exists
+      if (pendingFlush != null && !pendingFlush.isDone()) {
+        pendingFlush.cancel(false);
+      }
+      if (flushTimeoutMilliseconds > 0) {
+        pendingFlush =
+            executorService.schedule(this::flush, flushTimeoutMilliseconds, TimeUnit.MILLISECONDS);
+      }
     }
-  }
 
-  private void flush() {
-    // Cancel the existing scheduled task if it exists
-    if (pendingFlush != null && !pendingFlush.isDone()) {
-      pendingFlush.cancel(false);
+    private void flush() {
+      // Cancel the existing scheduled task if it exists
+      if (pendingFlush != null && !pendingFlush.isDone()) {
```

Comment:
I'm a bit uncomfortable with `flush` function having a side effect of cancelling a scheduler outside of flush function. can we make the flush function only about flushing and moving this cancellation outside?

Reply:
I pushed a commit that extracted it... Not sure I like it more though...